build/includes/website.mk

```diff
@@ -59,6 +76,7 @@ site-deploy-preview: site-static-preview
 
 site-test:
 	# generate actual html and run test against - provides a more accurate tests
+	$(MAKE) test-gen-api-docs
```

Comment:
I don't have a good answer for this - but because the website works off master - if we change our CRD data structures, that will change the website as well, and be out of sync until next release.

Only idea I've got - that in the advent of that happening - rename the old page and put an `expiryDate` of the next release in the toml.... but how would we get a publishDate in the new one, and still pass tests?

I'm just thinking out loud here, but any ideas?

Reply:
Tried to use custom short code in front-matter and it seems that variables are not supported there.
So currently I am thinking of combining one full generated html with `expiryVersion` shortcode. And new one with `publishVersion` shortcode in the same file.
And during test just compare the second half with `publishVersion`. And during release move bottom half to the top and generate new one at the bottom.

Reply:
That sounds fancy - but makes sense to me! I assume you can make that work? :smile:

Reply:
I have updated `gen-api-docs.sh` script. It now handles `base_version` environment variable.
What it does is get all lines before `publishVersion` line. Then extract what in the middle of this feature shortcode. Compares with what was generated if diff then do version compare.
Then set `expiryVersion` of old part with new `base_version` fresh generated doc now have new `publishVersion`.